spec/system/edit_work_spec.rb

```diff
@@ -72,8 +72,16 @@
 
     # Filling in related content
     find('.nav-link', text: 'Related content (optional)').click
-    fill_in('Link text', with: updated_related_links.first['text'])
-    fill_in('URL', with: updated_related_links.first['url'])
+    fill_in('Link text', with: 'delete')
+    fill_in('URL', with: 'me')
+    # Test adding a new nested field
+    click_link_or_button('+ Add another related link')
+    fill_in('work_related_links_attributes_1_text', with: updated_related_links.first['text'])
+    fill_in('work_related_links_attributes_1_url', with: updated_related_links.first['url'])
+    # Test removing a nested field
+    within('div[data-index="0"]') do
+      find('button[data-action="click->nested-form#delete"]').click
+    end
```

Comment:
Here's how I'm testing adding and removing nested fields, but note that there's a bit of false confidence because this system spec needs work. See this still-top-of-ready issue: https://github.com/sul-dlss/hungry-hungry-hippo/issues/138